llama-index-integrations/graph_stores/llama-index-graph-stores-falkordb/llama_index/graph_stores/falkordb/base.py

```diff
@@ -31,7 +31,11 @@ def __init__(
         self._node_label = node_label
 
         self._driver = FalkorDB.from_url(url).select_graph(database)
-        self._driver.query(f"CREATE INDEX FOR (n:`{self._node_label}`) ON (n.id)")
+
+        try:
+            self._driver.query(f"CREATE INDEX FOR (n:`{self._node_label}`) ON (n.id)")
+        except Exception as e:
```

Comment:
what's the specific exception that you get when the index still exists? Preferably there is another way to check if the index exists versus catching an error, however, if not then I think what would be better here is to catch/pass only the specific error raised when an index already exists, and then raise any other exceptions.

Reply:
This is the error "redis.exceptions.ResponseError: Attribute 'id' is already indexed", let me think, maybe I can find an appropriate way to fix this.

Reply:
Hi @nerdai I didn't find a good way to solve it, it looks like it's impossible to check if the index is already existing when creating the index, if I split the check and execute with several steps, there could be a concurrency(race) issue, I just put the specific error there, and added a todo comment.

Also I reverted the legacy updated.